config/routes.rb

```diff
@@ -44,4 +46,18 @@
       end
     end
   end
+
+  api_router = Proc.new do |env|
+    original_endpoint = env.dig("action_dispatch.request.path_parameters", :original_endpoint)
+
+    version = env.fetch("HTTP_APIVERSION") { API_VERSIONS.last }
```

Comment:
I think this should be ENV not env, if we use env and the users sends up a bad APIVERSION header then they'll be able to trigger the fail

Reply:
Actually it's probably better to set this in Rails.application.config rather than an env var, that way we have the api version in version control

Reply:
Yep, agree, this was just testing out the idea. Should definitely respond with a http error